azure-configurationservice/tests/test_azure_configuration_client.py

```diff
@@ -0,0 +1,601 @@
+        to_set_kv.tags = {"a": "b", "c": "d"}
+        set_kv = self.app_config_client.set_configuration_setting(to_set_kv)
+        # remove the new key value from the service
+        self.app_config_client.delete_configuration_setting(set_kv.key)
+        assert (
+            to_set_kv.key == set_kv.key
+            and None is set_kv.label
+            and to_set_kv.value == set_kv.value
+            and to_set_kv.content_type == set_kv.content_type
+            and to_set_kv.tags == set_kv.tags
+            and to_set_kv.etag != set_kv.etag
+        )
+
+    def test_set_existing_configuration_setting_label_etag(self):
+        sample_kv = copy(self.test_data.label1_data[-1])
+
+        # create a new key value into AzConfig service
+        sample_kv.key = "unit_test_key_set" + self.test_data.key_uuid
+        to_set_kv = self.app_config_client.add_configuration_setting(sample_kv)
+        to_set_kv.value = to_set_kv.value + "a"
+        to_set_kv.tags = {"a": "b", "c": "d"}
+        set_kv = self.app_config_client.set_configuration_setting(to_set_kv)
+        # remove the new key value from the service
+        self.app_config_client.delete_configuration_setting(
+            set_kv.key, label=set_kv.label
+        )
+        assert (
+            to_set_kv.key == set_kv.key
+            and to_set_kv.label == to_set_kv.label
+            and to_set_kv.value == set_kv.value
+            and to_set_kv.content_type == set_kv.content_type
+            and to_set_kv.tags == set_kv.tags
+            and to_set_kv.etag != set_kv.etag
+        )
+
+    def test_set_existing_configuration_setting_label_wrong_etag(self):
+        sample_kv = copy(self.test_data.label1_data[-1])
+
+        # create a new key value into AzConfig service
+        sample_kv.key = "unit_test_key_set" + self.test_data.key_uuid
+        to_set_kv = self.app_config_client.add_configuration_setting(sample_kv)
+        to_set_kv.value = to_set_kv.value + "a"
+        to_set_kv.tags = {"a": "b", "c": "d"}
+        to_set_kv.etag = "wrong etag"
+        with pytest.raises(CloudError):
+            self.app_config_client.set_configuration_setting(to_set_kv)
+        self.app_config_client.delete_configuration_setting(
+            to_set_kv.key, label=to_set_kv.label
+        )
+
+    def test_set_configuration_setting_no_label_etag(self):
+        to_set_kv = copy(self.test_data.no_label_data[-1])
+        to_set_kv.key = "unit_test_key_set" + self.test_data.key_uuid
+        with pytest.raises(CloudError):
+            self.app_config_client.set_configuration_setting(to_set_kv)
+
+    def test_set_configuration_setting_label_etag(self):
+        to_set_kv = copy(self.test_data.label1_data[-1])
+        to_set_kv.key = "unit_test_key_set" + self.test_data.key_uuid
+        with pytest.raises(CloudError):
+            self.app_config_client.set_configuration_setting(to_set_kv)
+
+    def test_set_configuration_setting_no_label_no_etag(self):
```

Comment:
instead of doing `no_etag` could you just create a new setting and set it?
It will make it easier to understand that the whole purpose of the test is to see that you can set a non-existing setting

Reply:
Accept. I'll review where I use "copy".